### PR TITLE
[mds-stream] Updates to support knative eventing.

### PR DESCRIPTION
**Pull request by avatarneil**
## PR Checklist

 - [x] simple searchable title - `[mds-db] Add PG env var`, `[config] Fix eslint config`
 - [ ] briefly describe the changes in this PR
 - [ ] mark as draft if should not be merged
 - [ ] write tests for all new functionality

## Impacts
- [ ] Provider
- [ ] Agency
- [ ] Audit
- [ ] Policy
- [ ] Compliance
- [x] Stream

### CODE REVIEW

File: packages/mds-stream/index.ts [hunk 1]
```diff
@@ -1,3 +1,4 @@
+/* eslint-disable import/export */
```
Comment (mdurling) [inline packages/mds-stream/index.ts]: Why do we have to disable?

File: packages/mds-stream/index.ts [hunk 2]
```diff
@@ -15,9 +16,36 @@
  */
 
 import logger from 'mds-logger'
+import { env } from 'process'
```
Comment (mdurling) [inline packages/mds-stream/index.ts]: Should be `const { env } = process`

File: packages/mds-stream/index.ts [hunk 3]
```diff
@@ -135,15 +167,25 @@ async function writeStreamBatch(stream: Stream, field: string, values: unknown[]
 
 // put basics of vehicle in the cache
 async function writeDevice(device: Device) {
+  if (env.SINK) {
+    return write('ai.lacuna.device', JSON.stringify(device))
```
Comment (mdurling) [inline packages/mds-stream/index.ts]: That 6-character sequence of letters should not appear.

File: packages/mds-stream/index.ts [hunk 4]
```diff
@@ -135,15 +167,25 @@ async function writeStreamBatch(stream: Stream, field: string, values: unknown[]
 
 // put basics of vehicle in the cache
 async function writeDevice(device: Device) {
+  if (env.SINK) {
+    return write('mds.device', JSON.stringify(device))
```
Comment (mdurling) [inline packages/mds-stream/index.ts]: Is there a reason that we can't use the same names? `DEVICE_INDEX_STREAM`?

File: packages/mds-stream/index.ts [hunk 5]
```diff
@@ -135,15 +167,25 @@ async function writeStreamBatch(stream: Stream, field: string, values: unknown[]
...
+  }
   return writeStream(DEVICE_INDEX_STREAM, 'data', device)
 }
 
 async function writeEvent(event: VehicleEvent) {
+  if (env.SINK) {
+    return write('mds.event', JSON.stringify(event))
+  }
   return Promise.all([DEVICE_RAW_STREAM, PROVIDER_EVENT_STREAM].map(stream => writeStream(stream, 'event', event)))
```
Comment (mdurling) [inline packages/mds-stream/index.ts]: Notice that events get written to 2 different Redis streams, but only one knative. I am not sure how the `provider:event` stuff translates to knative.

File: packages/mds-stream/index.ts [hunk 6]
```diff
@@ -17,8 +17,35 @@
...
+
+function getBinding() {
+  if (!binding) {
+    const config = {
+      method: 'POST',
+      url: env.SINK
+    }
+
+    // eslint-disable-next-line new-cap
+    return new Cloudevent.bindings['http-binary0.2'](config)
```
Comment (janedotx) [inline packages/mds-stream/index.ts]: Shouldn't there be a line like `binding = new Cloudevent.binds['http-binary0.2'](config)` to cache it?

Comment (avatarneil) [reply packages/mds-stream/index.ts]: Why not? Changed in 5c0527e

File: packages/mds-stream/index.ts [hunk 7]
```diff
@@ -135,15 +168,25 @@ async function writeStreamBatch(stream: Stream, field: string, values: unknown[]
 
 // put basics of vehicle in the cache
 async function writeDevice(device: Device) {
+  if (env.SINK) {
+    return write('mds.device', JSON.stringify(device))
```
Comment (mdurling) [inline packages/mds-stream/index.ts]: I think `write` is a pretty generic name here. Should we use `emitCloudEvent` or something more narrative, similar to how we call it `writeStream`?

**Pull request merged**